app/controllers/one_login_controller.rb

```diff
@@ -0,0 +1,89 @@
+class OneLoginController < ApplicationController
+  before_action :one_login_enabled
+  before_action :set_sentry_context, only: :callback
+
+  def callback
+    auth = request.env['omniauth.auth']
+    session[:onelogin_id_token] = auth&.credentials&.id_token
+    candidate = OneLoginUser.authentificate(auth)
+
+    sign_in(candidate, scope: :candidate)
+    candidate.update!(last_signed_in_at: Time.zone.now)
+
+    redirect_to candidate_interface_interstitial_path
+  rescue OneLoginUser::Error => e
+    session[:one_login_error] = e.message
+    redirect_to auth_onelogin_sign_out_path
+  end
+
+  def bypass_callback
+    one_login_user_bypass = OneLoginUserBypass.new(
+      token: request.env['omniauth.auth']&.uid,
+    )
+    candidate = one_login_user_bypass.authentificate
+
+    if one_login_user_bypass.valid? && candidate.present?
+      sign_in(candidate, scope: :candidate)
+      candidate.update!(last_signed_in_at: Time.zone.now)
+
+      redirect_to candidate_interface_interstitial_path
+    else
+      flash[:warning] = one_login_user_bypass.errors.full_messages.join('\n')
+      redirect_to candidate_interface_create_account_or_sign_in_path
+    end
+  end
+
+  def sign_out
+    id_token = session[:onelogin_id_token]
+    one_login_error = session[:one_login_error]
+    reset_session
+
+    session[:one_login_error] = one_login_error
+    if OneLogin.bypass?
+      redirect_to candidate_interface_create_account_or_sign_in_path
+    else
+      # Go back to one login to sign out the user on their end as well
+      redirect_to logout_onelogin(id_token), allow_other_host: true
+    end
+  end
+
+  def sign_out_complete
+    if session[:one_login_error].present?
+      Sentry.capture_message(session[:one_login_error], level: :error)
+      redirect_to internal_server_error_path
+    else
+      redirect_to candidate_interface_create_account_or_sign_in_path
+    end
+  end
+
+  def failure
+    session[:one_login_error] = "One login failure with #{params[:message]} " \
+                                "for onelogin_id_token: #{session[:onelogin_id_token]}"
+
+    redirect_to auth_onelogin_sign_out_path
+  end
+
+private
+
+  def one_login_enabled
+    return if FeatureFlag.active?(:one_login_candidate_sign_in)
+
+    redirect_to root_path
+  end
+
+  def set_sentry_context
+    Sentry.set_extras(
+      omniauth_hash: request.env['omniauth.auth'].to_h,
```

Comment:
Yes.

The issue we have is that we cannot identify the user who might have problems login in. We can't use the token/sub to find the user in one login. And in particular for the case where we have users with a one login already, trying to use another one login account, we won't know their email_address because we don't let them login in this case.

Basically, if a sentry exception is raised, seeing the omniauth hash would be very helpful in debugging the issue.

Reply:
This doesn't mean with every request we send the omniauth hash to sentry, just on exceptions

Reply:
We'll save the hash in a table on exceptions

Reply:
We will remove this and save the hash in a table on exception